src/lib/dialog/dialog.ts

```diff
@@ -26,6 +28,18 @@ export class MdDialog {
     return this._parentDialog ? this._parentDialog._openDialogs : this._openDialogsAtThisLevel;
   }
 
+  /** Subject for notifying the user that all open dialogs have finished closing. */
+  private _afterAllClosed = new Subject<void>();
+
+  /** Subject for notifying the user that a dialog has opened. */
+  private _afterOpen = new Subject<MdDialogRef<any>>();
```

Comment:
The `_afterAllClosed` and `_afterOpen` subjects need to delegate to the state of the `_parentDialog` if there is one (the same way that `_openDialogs` does. This is because the application may contain multiple `MdDialog` instances at different points in the tree.